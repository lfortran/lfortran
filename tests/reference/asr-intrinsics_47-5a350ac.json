{
    "basename": "asr-intrinsics_47-5a350ac",
    "cmd": "lfortran --indent --show-asr --no-color {infile} -o {outfile}",
    "infile": "tests/../integration_tests/intrinsics_47.f90",
    "infile_hash": "9896cd656e8493f42821310348e5a443eb41db3451965afe1031a8bd",
    "outfile": null,
    "outfile_hash": null,
    "stdout": "asr-intrinsics_47-5a350ac.stdout",
    "stdout_hash": "ed418e78321297f79c36f2ed75cf1ce9bbe26b9d920754897ce1ea22",
    "stderr": null,
    "stderr_hash": null,
    "returncode": 0
}