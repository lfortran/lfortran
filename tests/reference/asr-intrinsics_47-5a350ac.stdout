(TranslationUnit
    (SymbolTable
        1
        {
            intrinsics_47:
                (Program
                    (SymbolTable
                        2
                        {
                            a:
                                (Variable
                                    2
                                    a
                                    []
                                    Local
                                    (ArrayConstant
                                        [(LogicalConstant
                                            .true.
                                            (Logical 4 [])
                                        )
                                        (LogicalConstant
                                            .true.
                                            (Logical 4 [])
                                        )
                                        (LogicalConstant
                                            .true.
                                            (Logical 4 [])
                                        )]
                                        (Logical 4 [((IntegerConstant 1 (Integer 4 []))
                                        (IntegerConstant 3 (Integer 4 [])))])
                                        ColMajor
                                    )
                                    ()
                                    Save
                                    (Logical 4 [((IntegerConstant 1 (Integer 4 []))
                                    (IntegerConstant 3 (Integer 4 [])))])
                                    Source
                                    Public
                                    Required
                                    .false.
                                ),
                            l:
                                (Variable
                                    2
                                    l
                                    []
                                    Local
                                    (ArrayAll
                                        (ArrayConstant
                                            [(LogicalConstant
                                                .true.
                                                (Logical 4 [])
                                            )
                                            (LogicalConstant
                                                .true.
                                                (Logical 4 [])
                                            )]
                                            (Logical 4 [((IntegerConstant 1 (Integer 4 []))
                                            (IntegerConstant 2 (Integer 4 [])))])
                                            ColMajor
                                        )
                                        ()
                                        (Logical 4 [])
                                        (LogicalConstant
                                            .true.
                                            (Logical 4 [])
                                        )
                                    )
                                    ()
                                    Save
                                    (Logical 4 [])
                                    Source
                                    Public
                                    Required
                                    .false.
                                )
                        })
                    intrinsics_47
                    []
                    [(=
                        (Var 2 l)
                        (ArrayAll
                            (ArrayConstant
                                [(LogicalConstant
                                    .true.
                                    (Logical 4 [])
                                )
                                (LogicalConstant
                                    .false.
                                    (Logical 4 [])
                                )]
                                (Logical 4 [((IntegerConstant 1 (Integer 4 []))
                                (IntegerConstant 2 (Integer 4 [])))])
                                ColMajor
                            )
                            ()
                            (Logical 4 [])
                            (LogicalConstant
                                .false.
                                (Logical 4 [])
                            )
                        )
                        ()
                    )
                    (=
                        (Var 2 l)
                        (ArrayAll
                            (ArrayConstant
                                [(LogicalConstant
                                    .false.
                                    (Logical 4 [])
                                )]
                                (Logical 4 [((IntegerConstant 1 (Integer 4 []))
                                (IntegerConstant 1 (Integer 4 [])))])
                                ColMajor
                            )
                            ()
                            (Logical 4 [])
                            (LogicalConstant
                                .false.
                                (Logical 4 [])
                            )
                        )
                        ()
                    )
                    (=
                        (Var 2 l)
                        (ArrayAll
                            (ArrayConstant
                                [(LogicalConstant
                                    .true.
                                    (Logical 4 [])
                                )]
                                (Logical 4 [((IntegerConstant 1 (Integer 4 []))
                                (IntegerConstant 1 (Integer 4 [])))])
                                ColMajor
                            )
                            ()
                            (Logical 4 [])
                            (LogicalConstant
                                .true.
                                (Logical 4 [])
                            )
                        )
                        ()
                    )
                    (=
                        (Var 2 l)
                        (ArrayAll
                            (Var 2 a)
                            ()
                            (Logical 4 [])
                            ()
                        )
                        ()
                    )
                    (=
                        (Var 2 l)
                        (ArrayAll
                            (ArrayConstant
                                [(IntegerCompare
                                    (IntegerConstant 1 (Integer 4 []))
                                    Eq
                                    (IntegerConstant 2 (Integer 4 []))
                                    (Logical 4 [])
                                    (LogicalConstant
                                        .false.
                                        (Logical 4 [])
                                    )
                                )
                                (IntegerCompare
                                    (IntegerConstant 0 (Integer 4 []))
                                    Eq
                                    (IntegerConstant 0 (Integer 4 []))
                                    (Logical 4 [])
                                    (LogicalConstant
                                        .true.
                                        (Logical 4 [])
                                    )
                                )]
                                (Logical 4 [((IntegerConstant 1 (Integer 4 []))
                                (IntegerConstant 2 (Integer 4 [])))])
                                ColMajor
                            )
                            ()
                            (Logical 4 [])
                            (LogicalConstant
                                .false.
                                (Logical 4 [])
                            )
                        )
                        ()
                    )
                    (If
                        (Var 2 l)
                        [(ErrorStop
                            ()
                        )]
                        []
                    )]
                )
        })
    []
)
